lua/includes/modules/playerload.lua

```diff
@@ -0,0 +1,9 @@
+hook.Add( "PlayerInitialSpawn","FullLoadSetup", function( spawnedPly )
+    hook.Add( "SetupMove", spawnedPly, function( self, ply, _, cmd )
+        if self ~= ply then return end
+        if cmd:IsForced() then return end
+
+        hook.Run( "PlayerFullLoad", self )
+        hook.Remove( "SetupMove", self )
```

Comment:
should you call this before hook.Run in case hook.Run errors the hook will still be removed

Reply:
I don't think the `hook.Run` will ever error in this case?

Reply:
will hook.Run error if the whatever implements that hook errors? or does hook.Run catch the errors

Reply:
I haven't seen any big addons guard their `hook.Run`'s

Reply:
The wiki only gets the returned data so in a failing case it would just return `nil` no?

Reply:
if im not mistaken it calls the functions in the hook table until one of them returns values and then returns those values 

so if one of those hook callbacks errors then this errors unless hook.Run catches those errors

Reply:
Just don't write bad code 🤣 /s

Reply:
ye its probably fine because this hook is only called once 
all i was suggesting was changing the order of those 2 lines

Reply:
I do agree those lines should be swapped, hook.Run does not catch errors, and if it does error, it'll error on every SetupMove.